BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java

```diff
@@ -99,8 +100,26 @@ public void cancelEvent(int eventId){
         }
     }
 
-    public List<EventDTO> findByDateBetween(LocalDate rangeStart, LocalDate rangeEnd,EventTypes eventType) {//Pass in event type
-        return mapEventsToDtos(eventRepository.findByStartDateBetweenAndEventType(rangeStart, rangeEnd,new EventType(eventType.getEventTypeId())));
+    public List<String> rejectEvent(int eventId, String deniedMessage){
+        List<String> responses = new ArrayList<>();
+        Optional<Event> retrievedEvent = eventRepository.findById(eventId);
+        if (retrievedEvent.isPresent()) {
+            Event event = retrievedEvent.get();
+            if (event.getEventStatus().getEventStatusId() == EventStatuses.REJECTED.getEventStatusId()){
```

Comment:
Can we reject an approved event? Or an event in the past?

Reply:
Yes i can't see any reason why we can't?